src/libs/ajax/loaded-data/useLoadedData.test.ts

```diff
@@ -61,14 +62,15 @@ describe('useLoadedData hook', () => {
     const updateData = hookResult1[1];
 
     // Act
+    let updatePromise: Promise<void>;
     act(() => {
-      void updateData(async (): Promise<TestData> => {
+      updatePromise = updateData(async (): Promise<TestData> => {
```

Comment:
this is not as hacky as I feared... but lets get a comment in here that we are hanging onto the promise for later flow so that we can get a chance to record the "loading" state in between.... (same for any other tests like this)

Reply:
Updated these to use "controlled promises" in https://github.com/DataBiosphere/terra-ui/pull/4013/commits/bee002d5577e81b02bab3cd916b6ca3aa807ff7e. Also got rid of the delays.